assembly/runtime.ts

```diff
@@ -801,26 +842,37 @@ export class RootContext extends BaseContext {
    * @param timeout_milliseconds Timeout for the request, in milliseconds.
    * @param cb Callback to be invoked when the request is complete.
    */
-  httpCall(cluster: string, headers: Headers, body: ArrayBuffer, trailers: Headers,
-    timeout_milliseconds: u32, cb: HttpCallback): WasmResultValues {
-
+  httpCall(cluster: string, headers: Headers, body: ArrayBuffer, trailers: Headers, timeout_milliseconds: u32, origin_context_id: u32, cb: (origin_context_id:u32, headers: u32,  body_size: usize, trailers: u32) => void): WasmResultValues {
     let buffer = String.UTF8.encode(cluster);
     let header_pairs = serializeHeaders(headers);
     let trailer_pairs = serializeHeaders(trailers);
-    let token = globalU32Ref;
+    let token = new Reference<u32>();
+    this.http_calls_.set(token.data, new HttpCallback(origin_context_id, cb));
```

Comment:
refering to `token.data` can only happen proxy_http_call is called.
proxy_http_call returns a token using this pointer. this token is used later to co-relate the response.
see also my next comment.

Reply:
Oh, of course. 
So, I need to move line 850 to 854....